src/main.py

```diff
@@ -34,12 +83,13 @@ def datasets(
 
 
 @app.get("/datasets/{dataset_id}", response_model=Optional[schemas.Dataset])
-def dataset(
+async def dataset(
     request: Request,
     response: Response,
     dataset_id: str,
     metadata_store: StubMetadataStore = Depends(StubMetadataStore),
 ):
+    logger.info(f"Received request for dataset with ID: {dataset_id}", dataset_id)
```

Comment:
right idea but using the data field for application specific varibles, eg:

```
logger.info("Received request for dataset by id", data={"dataset_id": dataset_id})
```

the idea is that by recording variables like this with fields, when structured logs ends up in one great big log store (where they're searchable) you'll be able to search for logs dealing with a specifc (in this example) dataset id. i.e use fields by preference for variables as they're more searchable that interpolated strings.

I wont comment them all, but can you update the others in the same way please.